mechanics/RELEASE-VERSIONING-PRINCIPLES.md

```diff
@@ -0,0 +1,953 @@
+   <td><strong>API Endpoints</strong>
+   </td>
+   <td><strong>API Type Served</strong>
+   </td>
+   <td><strong>Availability</strong>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >0.19.x
+   </td>
+   <td rowspan="2" >2020-09-29
+   </td>
+   <td rowspan="2" >2021-03-16
+   </td>
+   <td rowspan="2" >1.1?.x
+   </td>
+   <td rowspan="2" >v1
+   </td>
+   <td rowspan="2" >N/A
+   </td>
+   <td>v1
+   </td>
+   <td>v1
+   </td>
+   <td>default
+   </td>
+  </tr>
+  <tr>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td><strong>not served</strong>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >0.18.x
+   </td>
+   <td rowspan="3" >2020-08-18
+   </td>
+   <td rowspan="3" >2021-02-02
+   </td>
+   <td rowspan="3" >1.1?.x
+   </td>
+   <td rowspan="3" >v1
+   </td>
+   <td rowspan="3" >Yes
+   </td>
+   <td>v1
+   </td>
+   <td>v1
+   </td>
+   <td>default
+   </td>
+  </tr>
+  <tr>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td>optional (deprecated)
+   </td>
+  </tr>
+  <tr>
+   <td>v1alpha1
+   </td>
+   <td>v1alpha1
+   </td>
+   <td><strong>removed</strong>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="3" >0.16.x
+   </td>
+   <td rowspan="3" >2020-07-07
+   </td>
+   <td rowspan="3" >2020-12-22
+   </td>
+   <td rowspan="3" >1.18.x
+   </td>
+   <td rowspan="3" >v1beta1
+   </td>
+   <td rowspan="3" >N/A
+   </td>
+   <td>v1
+   </td>
+   <td>v1
+   </td>
+   <td>optional
+   </td>
+  </tr>
+  <tr>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td>default
+   </td>
+  </tr>
+  <tr>
+   <td>v1alpha1
+   </td>
+   <td>v1alpha1
+   </td>
+   <td><strong>not served</strong>
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >0.15.x
+   </td>
+   <td rowspan="2" >2020-05-26
+   </td>
+   <td rowspan="2" >2020-11-10
+   </td>
+   <td rowspan="2" >1.16.x
+   </td>
+   <td rowspan="2" >v1beta1
+   </td>
+   <td rowspan="2" >Yes
+   </td>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td>default
+   </td>
+  </tr>
+  <tr>
+   <td>v1alpha1
+   </td>
+   <td>v1alpha1
+   </td>
+   <td>optional (deprecated)
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >0.14.x
+   </td>
+   <td rowspan="2" >2020-04-14
+   </td>
+   <td rowspan="2" >2020-09-29
+   </td>
+   <td rowspan="2" >1.16.x
+   </td>
+   <td rowspan="2" >v1beta1
+   </td>
+   <td rowspan="2" >N/A
+   </td>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td>default
+   </td>
+  </tr>
+  <tr>
+   <td>v1alpha1
+   </td>
+   <td>v1alpha1
+   </td>
+   <td>optional (deprecated)
+   </td>
+  </tr>
+  <tr>
+   <td rowspan="2" >0.13.x
+   </td>
+   <td rowspan="2" >2020-03-03
+   </td>
+   <td rowspan="2" >2020-08-18
+   </td>
+   <td rowspan="2" >1.15.x
+   </td>
+   <td rowspan="2" >v1alpha1
+   </td>
+   <td rowspan="2" >N/A
+   </td>
+   <td>v1beta1
+   </td>
+   <td>v1beta1
+   </td>
+   <td>optional
+   </td>
+  </tr>
+  <tr>
+   <td>v1alpha1
+   </td>
+   <td>v1alpha1
+   </td>
+   <td>default
+   </td>
+  </tr>
+</table>
+
+### Eventing:
+
+TODO
```

Comment:
Can we open issues for these TODOs so this doc doesn't rot?  cc @vaikas @lionelvillard @grantr @n3wscott @nachocano @matzew

Reply:
https://github.com/knative/eventing/issues/3006